src/PrestaShopBundle/Bridge/Smarty/ToolbarFlagsConfigurator.php

```diff
@@ -52,14 +54,29 @@ class ToolbarFlagsConfigurator implements ConfiguratorInterface
      */
     private $configuration;
 
+    /**
+     * @var Documentation
+     */
+    private $documentation;
+
     /**
      * @param RouterInterface $router
      * @param Configuration $configuration
      */
-    public function __construct(RouterInterface $router, Configuration $configuration)
+    public function __construct(RouterInterface $router, Configuration $configuration, ?Documentation $documentation = null)
     {
+        if (null === $documentation) {
+            @trigger_error(
+                'Not specifying the optional Documentation parameter is deprecated since version 8.0',
+                E_USER_DEPRECATED
+            );
+
+            $documentation = SymfonyContainer::getInstance()->get('prestashop.core.help.documentation');
```

Comment:
I might be missing context but here you need to get the container like a singleton? Cannot be injected?

Reply:
This is only for retro-compatibility, I've added changes in DI to inject dependency. But if you have overriden that class then, you won't have it, and the code should still be working.

Reply:
Good idea 👍 but it is not needed because this class was introduced in 8.0.0
https://github.com/PrestaShop/PrestaShop/commits/bf332c4a257eae630a5a9915378f39048df02569/src/PrestaShopBundle/Bridge/Smarty/ToolbarFlagsConfigurator.php
So it was never released and is not yet part of the Public API. Which means we can still break it 😄 it's like a `0.x` release

Reply:
Good news ! I'll break it then :D